src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8051,30 +8051,42 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
             }
 
             bool argIsIndex = argIsIndexNotRange.Value();
-            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-            if (!TryFindIndexOrRangeIndexerPattern(receiverOpt, receiverType, argIsIndex: argIsIndex,
-                out PropertySymbol? lengthOrCountProperty, out Symbol? patternSymbol, diagnostics, ref useSiteInfo))
+            var receiverValEscape = receiverOpt switch
+            {
+                BoundIndexOrRangeIndexerPatternReceiverPlaceholder { ValEscape: var valEscape } => valEscape,
+                BoundListPatternReceiverPlaceholder { ValEscape: var valEscape } => valEscape,
+                BoundSlicePatternReceiverPlaceholder { ValEscape: var valEscape } => valEscape,
+                _ => GetValEscape(receiverOpt, LocalScopeDepth)
+            };
+
+            var receiverPlaceholder = new BoundIndexOrRangeIndexerPatternReceiverPlaceholder(receiverOpt.Syntax, receiverValEscape, receiverOpt, receiverOpt.Type) { WasCompilerGenerated = true };
```

Comment:
>receiverOpt [](http://example.com/codeflow?start=132&length=11)

It looks like here we can create a placeholder on top of another placeholder. Feels strange.